collections/src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -428,8 +428,6 @@ class ArrayDeque[A] protected (
   override def copyToArray[B >: A](dest: Array[B], destStart: Int, len: Int): dest.type =
     copySliceToArray(srcStart = 0, dest = dest, destStart = destStart, maxItems = len)
 
-  override def className = "ArrayDeque"
```

Comment:
Why remove that?

Reply:
Because it's not needed. If you define it anyway all subclasses have to do the same (which they don't). Queue.toString used to print "ArrayDeque".

Reply:
I think that’s fine to define it in all subclasses.

Also, the scaladoc says:

> It is recommended to overwrite this method even if the default implementation returns the correct name, to avoid the implementation using reflection.

Which I agree with.

Reply:
The old collections library doesn't have that notice. The only use of this method is for debug output, it's never in a hot code path. The default implementation is perfectly fine for that. IMHO the recommendation should be not to override it unless necessary.

Reply:
OK then.